Updated $filter general explanation (#10156)
microsoftgraph/microsoft-graph-docs: concepts/query-parameters.md

Support for `$filter` operators varies across Microsoft Graph APIs. The following logical operators are generally supported: 

- equals `eq` / not equals `ne`
- less than `lt` / greater than `gt`

> - less or equals than `le` / greater or equals than `ge`

Reviewer: less than or equal to / greater than or equal to

- and `and` / or `or`
- in `in`
- Negation `not`
- lambda operator any `any`
- lambda operator all `all`
- Starts With `startsWith`
- Ends With `endsWith`

> **Note:** support for these operators varies by entity. Please read the specific entity documentation for details.
> The `contains` string operator is currently not supported on any Microsoft Graph resources.

For some usage examples, see the following table. For more details about `$filter` syntax, see the [OData protocol][odata-filter].  
The following table shows some examples that use the `$filter` query parameter.

> **Note:** Click the examples to try them in [Graph Explorer][graph-explorer].

| Description | Example
|:------------|:--------|

> | Search for users with the name Mary across multiple properties. | [`https://graph.microsoft.com/v1.0/users?$filter=startswith(displayName,'mary') or startswith(givenName,'mary') or startswith(surname,'mary') or startswith(mail,'mary') or startswith(userPrincipalName,'mary')`](https://developer.microsoft.com/graph/graph-explorer?request=users?$filter=startswith(displayName,'mary')+or+startswith(givenName,'mary')+or+startswith(surname,'mary')+or+startswith(mail,'mary')+or+startswith(userPrincipalName,'mary')&method=GET&version=v1.0) |

Author: ```suggestion
| Get all users with the name Mary across multiple properties. | [`https://graph.microsoft.com/v1.0/users?$filter=startswith(displayName,'mary') or startswith(givenName,'mary') or startswith(surname,'mary') or startswith(mail,'mary') or startswith(userPrincipalName,'mary')`](https://developer.microsoft.com/graph/graph-explorer?request=users?$filter=startswith(displayName,'mary')+or+startswith(givenName,'mary')+or+startswith(surname,'mary')+or+startswith(mail,'mary')+or+startswith(userPrincipalName,'mary')&method=GET&version=v1.0) |
```

> | Search for users with mail domain equal to 'hotmail.com' | [`https://graph.microsoft.com/v1.0/users?$count=true&$filter=endsWith(mail,'@hotmail.com')`](https://developer.microsoft.com/en-us/graph/graph-explorer?request=users%3F%24count%3Dtrue%26%24filter%3DendsWith(mail%2C'%40hotmail.com')%26%24select%3Did%2CdisplayName%2Cmail&method=GET&version=v1.0&GraphUrl=https://graph.microsoft.com&headers=W3sibmFtZSI6IkNvbnNpc3RlbmN5TGV2ZWwiLCJ2YWx1ZSI6ImV2ZW50dWFsIn1d) |

Author: ```suggestion
| Get all users with mail domain equal to 'hotmail.com' | [`https://graph.microsoft.com/v1.0/users?$count=true&$filter=endsWith(mail,'@hotmail.com')`](https://developer.microsoft.com/en-us/graph/graph-explorer?request=users%3F%24count%3Dtrue%26%24filter%3DendsWith(mail%2C'%40hotmail.com')%26%24select%3Did%2CdisplayName%2Cmail&method=GET&version=v1.0&GraphUrl=https://graph.microsoft.com&headers=W3sibmFtZSI6IkNvbnNpc3RlbmN5TGV2ZWwiLCJ2YWx1ZSI6ImV2ZW50dWFsIn1d) |
```
